src/matcha_ml/services/azure_service.py

```diff
@@ -98,36 +102,54 @@ def _get_principal_id(self) -> str:
 
         return user_object_id
 
-    def _check_required_role_assignments(self) -> bool:
-        """Check if the user has one of the required sets of roles.
-
-        Returns:
-            bool: True if required roles are assigned
+    def _fetch_user_roles(self) -> list[str]:
+        """Fetch the Azure roles for the user.
 
         Raises:
-            MatchaPermissionError: when the user does not have required roles
+            MatchaError: when the role assignments for a subscription can't be fetched from Azure, likely a connection issue.
 
+        Returns:
```

Comment:
Nit: should the docstring return information be stated first? Thinking about how this will get turned into documentation. Does 'list' need to be capitalised, or is that just a convention for typing?

Reply:
Return is always stated last as the docstring ordering reflects the natural ordering of the code, e.g., parameters come first because they're part of the function signature, raises is after because code being executed within the function raises errors, and return is last because it's the end of the function execution.